Lib/ufo2ft/featureWriters/kernFeatureWriter.py

```diff
@@ -424,10 +424,16 @@ def _makeKerningLookups(self):
         if shouldSplit:
             # make one DFLT lookup with script-agnostic characters, and two
             # LTR/RTL lookups excluding pairs from the opposite group.
-            # We drop kerning pairs with ambiguous direction.
+            # We drop kerning pairs with ambiguous direction: i.e. those containing
+            # glyphs from scripts with different overall horizontal direction, or
+            # glyphs with incompatible bidirectional type (arabic numbers vs
+            # arabic numerals).
             pairs = []
             for pair in self.context.kerning.pairs:
-                if "RTL" in pair.directions and "LTR" in pair.directions:
+                if ("RTL" in pair.directions and "LTR" in pair.directions) or (
+                    "R" in pair.bidiTypes
+                    and ("N" in pair.bidiTypes or "L" in pair.bidiTypes)
```

Comment:
What does N here stand for, neutral? Would this be true for punctuation as well? Numbers are weak L not neutral, real neutrals would be kerned RTL with RTL glyphs.

Reply:
I use "N" as a shorthand for the two bidi type that refers to numbers "EN" and "AN", both with a (weak) LTR direction.

https://github.com/googlefonts/ufo2ft/blob/6d89fe4611cb28b7e82c2110e984250412e61e0d/Lib/ufo2ft/featureWriters/kernFeatureWriter.py#L118-L134

neutrals (neither strong LTR nor strong RTL nor numbers) will not contribute to determine a kern pair's overal bidi type

Reply:
I see, then I’d simplify this and return L for numbers as well. We are only concerned with resolved directionality and that is either LTR or RTL and weak LTR will always resolve to LTR, so that is all we care about.

Reply:
makes sense. done